drivers/lora/CMakeLists.txt

```diff
@@ -4,4 +4,5 @@ zephyr_library_named(loramac-node)
 
 zephyr_library_sources_ifdef(CONFIG_LORA_SHELL shell.c)
 zephyr_library_sources_ifdef(CONFIG_HAS_SEMTECH_RADIO_DRIVERS hal_common.c)
+zephyr_library_sources_ifdef(CONFIG_HAS_SEMTECH_RADIO_DRIVERS sx12xx_common.c)
```

Comment:
How different is this when compared to the other common driver? Had you made sure that `hal_common.c` is common to all drivers and this one is just for sx12xx? If not sure, I'd recommend to just split the common functionality into `sx12xx_common.c` file for now. If we face any requirement to split further in future, we'll do that by that time.

Reply:
The difference is that hal_common.c contains all of the timer and board callbacks from the library, while `sx12xx_common.c` contains the Zephyr API implementation that uses the Radio HAL in LoRaMAC. I believe some of the stuff that went into `hal_common.c` might actually be needed even if we just use the LoRaWAN functionality in the library and implement our own radio drivers.

I don't particularly care about the naming here, but I figured it might be useful to separate the board stuff from the API implementation. How about calling the files `hal_board.c` and `hal_radio.c` to make the distinction clearer?

Reply:
> The difference is that hal_common.c contains all of the timer and board callbacks from the library, while `sx12xx_common.c` contains the Zephyr API implementation that uses the Radio HAL in LoRaMAC. I believe some of the stuff that went into `hal_common.c` might actually be needed even if we just use the LoRaWAN functionality in the library and implement our own radio drivers.

Okay. Let's keep it as it is.